src/gui/src/tclCmdInputWidget.cpp

```diff
@@ -685,16 +685,36 @@ void TclCmdInputWidget::executeCommand(const QString& cmd,
   } else {
     if (!is_ok) {
       // Show output on error despite silent
-      processTclResult(is_ok);
+      processTclResult(return_code);
     }
   }
 
   emit commandFinishedExecuting(is_ok);
 }
 
-void TclCmdInputWidget::processTclResult(bool is_ok)
+void TclCmdInputWidget::processTclResult(const int tcl_result)
 {
+  const bool is_ok = (tcl_result == TCL_OK);
   emit addResultToOutput(Tcl_GetString(Tcl_GetObjResult(interp_)), is_ok);
+
+  if (!is_ok) {
+    // Tcl_GetReturnOptions returns an object with a ref count of 0.
+    // We DO NOT own it and MUST NOT decrement its ref count.
+    Tcl_Obj* options = Tcl_GetReturnOptions(interp_, tcl_result);
+
+    // Create a key to look up the stack trace in the options dictionary.
+    Tcl_Obj* key = Tcl_NewStringObj("-errorinfo", -1);  // refCount is now 1
+
+    // Look up the stack trace.
+    Tcl_Obj* stackTrace = nullptr;
+    if (Tcl_DictObjGet(nullptr, options, key, &stackTrace) == TCL_OK
+        && stackTrace) {
+      const char* result_msg = Tcl_GetString(stackTrace);
+      emit addTextToOutput(QString::fromStdString(result_msg), Qt::red);
```

Comment:
I don't think you need this call `QString::fromStdString` since it's starting as a `const char*`. Above the output of `Tcl_GetString` is passed directly to `addResultToOutput`